crates/node/service/src/actors/derivation.rs

```
@@ -202,6 +233,18 @@ where
 
                     self.process(InboundDerivationMessage::NewDataAvailable).await?;
                 }
+                signal = self.derivation_signal_rx.recv() => {
+                    let Some(signal) = signal else {
+                        error!(
+                            target: "derivation",
+                            ?signal,
+                            "DerivationActor failed to receive signal"
+                        );
+                        return Ok(());
```

Should we return an error in that case (or the one above)?

---

I think you're right we should error here since it's critical to consensus